### PR TITLE
ZXS-1758 onlyRelated attribute added

### PR DESCRIPTION
**Pull request by CMazzilli**
ZXS-1758 fix

### CODE REVIEW

**Comment by CMazzilli** [comment]
test added

**Comment by gordyt** [comment]
@CMazzilli is ZXS-1758 from your Jira instance?  I can't seem to find that project in either of our Jira installs.  Might I trouble you to add the details from that ticket to this PR description so we can git a bit of context?  Thanks!!

**Comment by CMazzilli** [comment]
The problem described into the issue was that `mailbox.log` logs the exception :
```
com.zimbra.common.service.ServiceException: permission denied: need right: countDomain for global globalacltarget
ExceptionId:qtp1816147548-1637:https:https://172.18.0.2:9071/service/admin/soap/BatchRequest:1529065339620:93781a3657b5a056
Code:service.PERM_DENIED
        at com.zimbra.common.service.ServiceException.PERM_DENIED(ServiceException.java:318)
        at com.zimbra.cs.service.admin.AdminAccessControl$ACLAccessControl.checkRight(AdminAccessControl.java:811)
        at com.zimbra.cs.service.admin.AdminDocumentHandler.checkRight(AdminDocumentHandler.java:827)
        at com.zimbra.cs.service.admin.AdminDocumentHandler.checkRight(AdminDocumentHandler.java:800)
        at com.zimbra.cs.service.admin.CountObjects.handle(CountObjects.java:149)
        at com.zimbra.soap.SoapEngine.dispatchRequest(SoapEngine.java:623)
        at com.zimbra.soap.SoapEngine.dispatch(SoapEngine.java:456)
        at com.zimbra.soap.SoapEngine.dispatch(SoapEngine.java:274)
        at com.zimbra.soap.SoapServlet.doWork(SoapServlet.java:303)
        at com.zimbra.soap.SoapServlet.doPost(SoapServlet.java:213)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
        at com.zimbra.cs.servlet.ZimbraServlet.service(ZimbraServlet.java:206)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:821)
```
when a domain admin logs into the admin console.

Also we have noticed that the counter of the domains shown in the main page of the admin console is always `0`.

Such domain admin has not the `countDomain` right.
The proposed solution in PR return **the only related** domain count instead of 0.

File: store/src/java/com/zimbra/cs/service/admin/CountObjects.java [hunk 1]
```diff
@@ -16,9 +16,7 @@
  */
 package com.zimbra.cs.service.admin;
 
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/admin/CountObjects.java]: Can we add specific imports

File: store/src/java/com/zimbra/cs/service/admin/CountObjects.java [hunk 2]
```diff
@@ -144,6 +144,23 @@ public Element handle(Element request, Map<String, Object> context)
                 checkDomainRight(zsc, domain, getAssociatedRight(countObjectsType));
                 count += prov.countObjects(countObjectsType, domain, ucService);
             }
+        } else if (countObjectsType.equals(CountObjectsType.domain) && (zsc.getAuthToken().isDelegatedAdmin() || zsc.getAuthToken().isDomainAdmin() ) && req.getOnlyRelated())
+        {
+            RightCommand.Grants grants = prov.getGrants(null, null, null, "usr", GranteeSelector.GranteeBy.id, zsc.getAuthtokenAccountId(), false);
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/admin/CountObjects.java]: PLease use the constants GranteeTYpe.GT_USR

File: store/src/java/com/zimbra/cs/service/admin/CountObjects.java [hunk 3]
```diff
@@ -144,6 +144,23 @@ public Element handle(Element request, Map<String, Object> context)
...
+            if (grants != null)
+            {
+                Set<RightCommand.ACE> acEs      = grants.getACEs();
+                Set<String>           domainIds = new HashSet<String>();
+                for (RightCommand.ACE acE : acEs)
+                {
+                    if (acE.targetType().equals("domain") && !domainIds.contains(acE.targetId()))
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/admin/CountObjects.java]: Please use constants TargetType

**Pull request merged**